### PR TITLE
RDF test migration (part 3)

### PR DESCRIPTION
**Pull request by captainsafia**
- Fix handling of struct types in JSON body
- Migrate tests related to special parameter types (`HttpContext`, `ClaimsPrincipal`, etc) to shared infra
- Migrate tests related to endpoint filters to shared infra
- Use `await` in async-enabled void-returning filtered handlers

### CODE REVIEW

File: src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs [hunk 1]
```diff
@@ -112,21 +114,8 @@ private static async ValueTask<(bool, T?)> TryResolveBodyAsync<T>(HttpContext ht
...
-                        }
-                        else
-                        {
-                            logOrThrowExceptionHelper.ImplicitBodyNotProvided(parameterName);
-                        }
-                        httpContext.Response.StatusCode = StatusCodes.Status400BadRequest;
-                        return (false, bodyValue);
-                    }
-                    return (true, bodyValue);
+                    bodyValue = await httpContext.Request.ReadFromJsonAsync<T>();
```
Comment (BrennanConroy) [inline src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs:117]: https://github.com/dotnet/aspnetcore/pull/48269#pullrequestreview-1438729332

Tracked anywhere?

File: src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Filters.cs [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Http.Generators.Tests;
+
+public abstract partial class RequestDelegateCreationTests : RequestDelegateCreationTestBase
+{
+    [Fact]
+    public async Task RequestDelegateInvokesFiltersButNotHandler_OnArgumentError()
```
Comment (BrennanConroy) [inline src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Filters.cs:15]: "InvokesFilters", maybe I'm missing it, where is this asserted?

File: src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Filters.cs [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+
+        await endpoint.RequestDelegate(httpContext);
+
+        // Assert
+        Assert.Null(httpContext.Items["invoked"]);
+        await VerifyResponseBodyAsync(httpContext, string.Empty, StatusCodes.Status400BadRequest);
+    }
+
+    [Fact]
+    public async Task RequestDelegateInvokesFilters_OnDelegateWithTarget()
```
Comment (BrennanConroy) [inline src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Filters.cs:41]: This one also doesn't look like it asserts the filter is called.

**Review by BrennanConroy — APPROVED** [review]

File: src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs (hunk 1 above)
Comment (captainsafia) [reply src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs:117]: I've opened https://github.com/dotnet/aspnetcore/pull/48573 to follow-up on this.

**Pull request merged**